### PR TITLE
doc: Fix create index syntax

### PR DESCRIPTION
**Pull request by jkosh44**
### Motivation
Fixes docs

### Checklist

- [X] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [X] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  - N/A

### CODE REVIEW

**Comment by jkosh44** [comment]
I'll rebase and update after https://github.com/MaterializeInc/materialize/pull/15370 is merged.

**Review by morsapaes — APPROVED** [review]
Thanks again for catching this, @jkosh44! After the rebase, only `sql-grammar.bnf` + the two `.svg` files should actually have changed; and those look good to me!

File: doc/user/content/sql/create-source/load-generator.md [hunk 1]
```diff
@@ -53,7 +53,7 @@ interval, the next number in the sequence is emitted.
 
 The auction load generator simulates an auction house, where users are bidding
 on an ongoing series of auctions. The auction source is meant to be used
-with [`CREATE VIEWS`](/sql/create-views), which will create the following five
+with , which will create the following five
```
Comment (mjibson) [inline doc/user/content/sql/create-source/load-generator.md:56]: missing word! I think you need `FOR ALL TABLES` here? This should be added to the load generator bnf as well.

Comment (jkosh44) [reply doc/user/content/sql/create-source/load-generator.md:56]: This is going to get fixed https://github.com/MaterializeInc/materialize/pull/15370 and I'm just going to overwrite this with that commit whenever it merges.

**Comment by jkosh44** [comment]
@morsapaes Any idea why the `select-stmt.svg` file changed even though I didn't change it's grammer?

**Pull request merged**